### ISSUE TITLE
Consider using Microsoft.Extensions.AI

### ISSUE THREAD

**Issue by stephentoub** [KeyserDSoze/Rystem#5]
The PlayFramework exposes an IChatClient interface. It's very similar to the IChatClient interface from Microsoft.Extensions.AI:
https://github.com/dotnet/extensions/blob/cfed375f3161f2e553e946b4f968b818e8e858f1/src/Libraries/Microsoft.Extensions.AI.Abstractions/ChatCompletion/IChatClient.cs#L26
If your framework used that core .NET interface as the exchange type, then other implementations of the interface would "just work" (there are already a bunch on nuget).

Similarly, the ChatClient type needn't be tied specifically to OpenAI. With minor tweaks, it can be made to work with any IChatClient, including but not limited to one for OpenAI.

Some background:

https://www.youtube.com/watch?v=qcp6ufe_XYo

https://devblogs.microsoft.com/dotnet/introducing-microsoft-extensions-ai-preview/

Thanks!

**Comment by KeyserDSoze** [comment]
Yes, I know the interface, but for now I want to create the integration specifically for OpenAi. The principle behind this choice is very simple, with a standard interface I can't release all the features that OpenAi has, and I don't want to have a bad design or less features because the competitors are struggling to catch up it. Furthermore, I've moved the playframework project with an entire new integration in [Rystem.OpenAi project](https://github.com/KeyserDSoze/Rystem.OpenAi).
